test/utils/initMatchers.js

```diff
@@ -151,3 +152,137 @@ chai.use((chaiAPI, utils) => {
     new chai.Assertion(this._obj).to.be.visible;
   });
 });
+
+chai.use((chaiAPI, utils) => {
+  function addConsoleMatcher(matcherName, methodName) {
+    /**
+     * @param {string[]} expectedMessages
+     */
+    function matcher(expectedMessages = []) {
+      // documented pattern to get the actual value of the assertion
+      // eslint-disable-next-line no-underscore-dangle
+      const callback = this._obj;
+
+      if (process.env.NODE_ENV !== 'production') {
```

Comment:
What's the motivation for a different assertion logic if node env is production?

Reply:
In prod it shouldn't match any console calls because they are stripped out. At least so far we don't have any console calls that are kept in prod bundles. Though we don't run tests on the prod bundle so it's not doing anything. Just implementing the semantics of a "to error in development" matcher.